### PR TITLE
add new seeds with traits

### PR DESCRIPTION
**Pull request by Kasvit**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [ ] specs added
- [ ] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [x] seed.rb file is updated if needed
- [ ] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -32,8 +32,75 @@
...
+      end
+    end
+  end
+
+  trait :comments_for_projects do
+    after(:create) do |user|
+      user.workspaces.each do |workspace|
+        workspace.projects.each do |project|
+          project.update(users: [user])
+          FactoryBot.create_list(:comment, 1.upto(5).to_a.sample, :for_project, user: user, commentable: project)
```
Comment (igorkasyanchuk) [inline spec/factories/users.rb]: this code 1.upto(5).to_a.sample is bad
rework

File: spec/factories/users.rb [hunk 2]
```diff
@@ -32,8 +32,75 @@
...
+        workspace.projects.each do |project|
+          FactoryBot.create(:task, :future, project: project)
+          FactoryBot.create(:task, :expired, project: project)
+          FactoryBot.create(:task, :completed, :expired, project: project)
+        end
+      end
+    end
+  end
+
+  trait :comments_for_tasks do
```
Comment (igorkasyanchuk) [inline spec/factories/users.rb]: be consistent with names of traits add "with_"

File: spec/factories/users.rb [hunk 3]
```diff
@@ -32,8 +32,75 @@
...
+      end
+    end
+  end
+
+  trait :watchers do
+    after(:create) do |user|
+      user.workspaces.each do |workspace|
+        workspace.projects.each do |project|
+          project.tasks.each do |task|
+            task.update(watchers: [user]) if [true, false].sample
```
Comment (igorkasyanchuk) [inline spec/factories/users.rb]: why we have [true, false].sample if trait "with_watchers" ? always create with watchers

File: spec/factories/users.rb [hunk 4]
```diff
@@ -32,8 +32,75 @@
...
     password { Faker::Internet.password }
-    role { Faker::Job.title }
+    role { "user" }
     department { Faker::Job.field }
     about { Faker::Lorem.sentences }
   end
-end
+
+  trait :admin do
+    role { "admin" }
```
Comment (igorkasyanchuk) [inline spec/factories/users.rb:42]: I think we have this field somewhere on UI

so we need to remove it from UI forms (in profile)

or rename in profile to "job_role"

File: spec/factories/users.rb (hunk 1 above)
Comment (romaslmd) [reply spec/factories/users.rb]: for the traits - add a controllable static value of number of comments, could be very useful later.. 
tip: `transient + evaluator`

**Review by igorkasyanchuk — APPROVED** [review]

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**